Assets/LeapMotion/Scripts/DetectionUtilities/ExtendedFingerDetector.cs

```diff
@@ -0,0 +1,110 @@
+﻿using UnityEngine;
+using System.Collections;
+using Leap;
+
+namespace Leap.Unity {
+
+  public class ExtendedFingerDetector : Detector {
+    [Tooltip("The interval in seconds at which to check this detector's conditions.")]
+    public float Period = .1f; //seconds
+
+    public IHandModel handModel = null;
+  
+    public PointingState Thumb = PointingState.Either;
+    public PointingState Index = PointingState.Either;
+    public PointingState Middle = PointingState.Either;
+    public PointingState Ring = PointingState.Either;
+    public PointingState Pinky = PointingState.Either;
+  
+    void Start () {
+      if(handModel == null){
+        handModel = gameObject.GetComponentInParent<IHandModel>();
```

Comment:
Start happens after OnEnable, so if the hand model is not assigned, the coroutine will give an error since it tries to access handModel which might be null.